mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageConverterTest.java

```diff
@@ -151,6 +152,18 @@ void convertShouldReturnCorrespondingDeletedMessageWhenNoRecipient() throws Exce
             .isEmpty();
     }
 
+    @Test
+    void convertShouldReturnCorrespondingDeletedMessageWhenInvalidRecipient() throws Exception {
+        MessageBuilder builder = getMessageBuilder();
+        builder.headers.remove(TO_FIELD);
+        builder.headers.remove(CC_FIELD);
+        builder.header(TO_FIELD, "bad@bad@bad");
+
+        DeletedMessage deletedMessage = deletedMessageConverter.convert(DELETED_MESSAGE_METADATA, buildMessage(builder, NO_ATTACHMENT));
+        assertThat(deletedMessage.getRecipients())
+            .isEmpty();
```

Comment:
Just curious... What happens if there is other CC / To headers? Are they preserved?

Reply:
added test